src/core.jl

```diff
@@ -0,0 +1,70 @@
+function reproject(input_data, output_projection; shape_out = nothing, order = 1, hdu_in = 1, hdu_out = 1, return_footprint = true)
+    if input_data isa ImageHDU
+        array_in, wcs_out = parse_input_data(input_data)
+    else
+        array_in, wcs_out = parse_input_data(input_data, hdu_in)
+    end
+    
+    if output_projection isa FITS || output_projection isa String
+        wcs_in, shape_out = parse_output_projection(output_projection, hdu_out)
+    else
+        wcs_in, shape_out = parse_output_projection(output_projection, shape_out)
+    end
+    
+    type_in = wcs_to_celestial_frame(wcs_in)
+    type_out = wcs_to_celestial_frame(wcs_out)
+    
+    if type_in == type_out
+        return array_in
+    end
+    
+    img_out = fill!(zeros(shape_out), NaN)
```

Comment:
Are you sure it's a good idea to initialise the matrix to NaN?  Why not zeros or even `undef`?  As it is now you're setting the memory twice, which is a huge waste of computation power, as you're going to set all elements anyway later, aren't you?

Reply:
Would `fill(NaN, shape_out)` be good?
I need `NaN` filled arrays to form the footprint of output image as done in python version.
I can't take any integer filled number as base as it will not form the correct grayscaled image.